<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EMControl</class>
 <widget class="QWidget" name="EMControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>387</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="paramLayout">
     <item row="0" column="1">
      <widget class="QSpinBox" name="EMIterationsSpinBox">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="value">
        <number>15</number>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="EMIterationsTextLabel">
       <property name="text">
        <string>EM iterations:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="EMToleranceTextLabel">
       <property name="text">
        <string>EM tolerance:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QDoubleSpinBox" name="EMToleranceSpinBox">
       <property name="decimals">
        <number>10</number>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.000001000000000</double>
       </property>
       <property name="value">
        <double>0.000001000000000</double>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="BPIterationsTextlabel">
       <property name="text">
        <string>BP iterations:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSpinBox" name="BPIterationsSpinBox">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100000</number>
       </property>
       <property name="value">
        <number>10000</number>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="BPToleranceTextLabel">
       <property name="text">
        <string>BP tolerance:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QDoubleSpinBox" name="BPToleranceSpinBox">
       <property name="decimals">
        <number>10</number>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_8">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="runEMPushButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Run EM</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="showLayout">
     <item row="0" column="0">
      <widget class="QCheckBox" name="showEMCheckBox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Show EM</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
  <action name="testAction">
   <property name="text">
    <string>Test</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>showEMCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>EMControl</receiver>
   <slot>showEMToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BPIterationsSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>EMControl</receiver>
   <slot>BPIterationsChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BPToleranceSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>EMControl</receiver>
   <slot>BPToleranceChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>EMIterationsSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>EMControl</receiver>
   <slot>EMIterationsChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>EMToleranceSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>EMControl</receiver>
   <slot>EMToleranceChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runEMPushButton</sender>
   <signal>pressed()</signal>
   <receiver>EMControl</receiver>
   <slot>runEMPushed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>165</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showEMToggled(bool)</slot>
  <slot>BPIterationsChanged(int)</slot>
  <slot>BPToleranceChanged(double)</slot>
  <slot>EMIterationsChanged(int)</slot>
  <slot>EMToleranceChanged(double)</slot>
  <slot>runEMPushed()</slot>
 </slots>
</ui>
